### PR TITLE
Add ASO role assignment to SDS

### PR DESCRIPTION
**Pull request by reespozzi**
This change:
- Adds ASO to SDS in demo and preview, similar to CFT

Similar to recent CFT https://github.com/hmcts/aks-cft-deploy/pull/389/files#diff-4ced5165d1c7ac5b6c06db4edad8405844a7a60a8a6c7423c4ec011a5f330aefR103 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: components/05-mis/03-interpolated-defaults.tf [hunk 1]
```diff
@@ -38,3 +38,8 @@ data "azurerm_key_vault" "acme" {
   name                = "acmedtssds${var.env}"
   resource_group_name = data.azurerm_resource_group.platform-rg.name
 }
+data "azurerm_subscription" "subscription" {}
+data "azurerm_user_assigned_identity" "aks" {
+  name                = "aks-${var.env}-mi"
+  resource_group_name = data.azurerm_resource_group.genesis_rg.name
+}
```
Comment (timja) [inline components/05-mis/03-interpolated-defaults.tf:45]: this is created as a resource in the same component you don't need a data source:
https://github.com/hmcts/aks-sds-deploy/blob/59775a12e027578968cce4b87ddbae843bd26889/components/05-mis/10-managed-identity.tf#L1-L7

File: environments/05-mis/ithc.tfvars [hunk 2]
```diff
@@ -0,0 +1 @@
+enable_debug = "true"
```
Comment (timja) [inline environments/05-mis/ithc.tfvars:1]: what's with this enable_debug?

**Review by timja — CHANGES_REQUESTED** [review]

File: environments/05-mis/ithc.tfvars (hunk 2 above)
Comment (reespozzi) [reply environments/05-mis/ithc.tfvars]: Removed now, was from copy&paste from another components tf vars to get all the files

File: environments/05-mis/dev.tfvars [hunk 3]
```diff
@@ -0,0 +1,2 @@
+enable_debug = "true"
```
Comment (timja) [inline environments/05-mis/dev.tfvars:1]: what's with this?

File: environments/05-mis/demo.tfvars [hunk 4]
```diff
@@ -0,0 +1,2 @@
+enable_debug = "true"
```
Comment (timja) [inline environments/05-mis/demo.tfvars:1]: what's with this?

**Review by timja — APPROVED** [review]

**Pull request merged**